src/Build/BackEnd/Components/Logging/LoggingService.cs

```diff
@@ -1589,8 +1591,17 @@ private void RouteBuildEvent(BuildEventArgs eventArg)
         /// <exception cref="Exception">Any exception which is a ExceptionHandling.IsCriticalException will not be wrapped</exception>
         private void InitializeLogger(ILogger logger, IEventSource sourceForLogger)
         {
+            ILogger UnwrapLoggerType(ILogger log)
+            {
+                return (log is ProjectCollection.ReusableLogger reusableLogger)
+                    ? reusableLogger.OriginalLogger
```

Comment:
Is it possible for this type of nesting to be more than one level deep? Often for this type of problem there is an extension method defined called `Unwrap` that recursively calls itself to handle arbitrary nesting scenarios.